playground/models/networks/encoder.py

```diff
@@ -127,10 +129,22 @@ def __init__(self, data_dims, noise_dim, latent_dims, network_architecture='synt
         latent_space, features = [], []
         inputs = [Input(shape=(dim,), name="enc_data_input_{}".format(i)) for i, dim in enumerate(data_dims)]
         standard_normal_sampler = Lambda(sample_standard_normal_noise, name='enc_normal_sampler')
-        if network_architecture == 'synthetic':
+        if noise_mode == 'add':
+            standard_normal_sampler.arguments = {'seed': config['seed'], 'noise_dim': noise_dim, 'mode': 'add'}
+        elif noise_mode == 'concat':
+            if network_architecture == 'mnist':
+                assert ((noise_dim % sqrt(data_dims[0]) == 0) & (noise_dim % sqrt(data_dims[1]) == 0)), \
+                'Expected to receive a noise_dim that can form a rectangle with the given inputs_dims. Received {} as noise' \
+                'and {} and {} for the data dimension.'.format(noise_dim, data_dims[0], data_dims[1])
             standard_normal_sampler.arguments = {'seed': config['seed'], 'noise_dim': noise_dim, 'mode': 'concatenate'}
+        elif noise_mode == 'product':
+            assert data_dims[0] == noise_dim & data_dims[1] == noise_dim, \
+            'Expected to receive a noise_dim that is equal to the given inputs_dims. Received {} as noise' \
+            'and {} and {} for the data dimension.'.format(noise_dim, data_dims[0], data_dims[1])
+            standard_normal_sampler.arguments = {'seed': config['seed'], 'noise_dim': noise_dim, 'mode': 'product'}
         else:
-            standard_normal_sampler.arguments = {'seed': config['seed'], 'noise_dim': noise_dim, 'mode': 'add'}
+            raise NotImplementedError("Only the noise modes 'add', 'concat' and 'product' are available.")
```

Comment:
You should raise a `ValueError` instead of `NotImplementedError` as the former is used to indicate invalid value passed to an argument and the latter is to indicate the there should be something which is yet to come.